include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -8,26 +8,38 @@
 #include "fiction/algorithms/path_finding/a_star.hpp"
 #include "fiction/algorithms/path_finding/cost.hpp"
 #include "fiction/algorithms/path_finding/distance.hpp"
+#include "fiction/algorithms/physical_design/wiring_reduction.hpp"
 #include "fiction/layouts/bounding_box.hpp"
+#include "fiction/layouts/clocking_scheme.hpp"
 #include "fiction/layouts/obstruction_layout.hpp"
 #include "fiction/traits.hpp"
-#include "fiction/utils/name_utils.hpp"
-#include "fiction/utils/placement_utils.hpp"
 
 #include <mockturtle/traits.hpp>
 #include <mockturtle/utils/stopwatch.hpp>
 
 #include <algorithm>
 #include <cmath>
 #include <cstdint>
-#include <tuple>
-#include <unordered_map>
+#include <optional>
+#include <ostream>
 #include <unordered_set>
+#include <utility>
 #include <vector>
 
 namespace fiction
 {
 
+/**
+ * Parameters for the post-layout optimization algorithm.
+ */
+struct post_layout_optimization_params
+{
+    /**
+     * Maximum number of relocations to try for each gate. Defaults to the number of tiles in a layout if not specified.
+     */
+    std::optional<uint64_t> max_gate_relocations;
```

Comment:
```suggestion
    std::optional<uint64_t> max_gate_relocations = std::nullopt;
```
For dedicated initialization.